src/patternfly/components/DataList/data-list.scss

```diff
@@ -132,19 +140,38 @@
   --pf-c-data-list--m-compact__cell--cell--MarginRight: var(--pf-global--spacer--md);
   --pf-c-data-list--m-compact__item-control--PaddingTop: var(--pf-global--spacer--sm);
   --pf-c-data-list--m-compact__item-control--PaddingBottom: 0;
+  --pf-c-data-list--m-compact__item-control--PaddingBottom: var(--pf-global--spacer--sm);
   --pf-c-data-list--m-compact__item-control--MarginRight: var(--pf-global--spacer--md);
   --pf-c-data-list--m-compact__item-action--PaddingTop: var(--pf-global--spacer--sm);
   --pf-c-data-list--m-compact__item-action--PaddingBottom: var(--pf-global--spacer--sm);
   --pf-c-data-list--m-compact__item-action--MarginLeft: var(--pf-global--spacer--md);
+  --pf-c-data-list--m-compact__item-action__action--MarginTop: calc(var(--pf-c-data-list--m-compact__item-action--PaddingTop) * -1);
+  --pf-c-data-list--m-compact__item-action__action--MarginBottom: calc(var(--pf-c-data-list--m-compact__item-action--PaddingBottom) * -1);
   --pf-c-data-list--m-compact__item-content--PaddingBottom: var(--pf-global--spacer--sm);
 
   @include pf-t-light;
 
+  font-size: var(--pf-c-data-list--FontSize);
+  line-height: var(--pf-c-data-list--LineHeight);
   overflow-wrap: break-word;
   list-style-type: disc;
   border-top: var(--pf-c-data-list--BorderTopWidth) solid var(--pf-c-data-list--BorderTopColor);
 
   &.pf-m-compact {
+    --pf-c-data-list__check--FontSize: var(--pf-c-data-list--m-compact__check--FontSize);
+    --pf-c-data-list__action--MarginTop: var(--pf-c-data-list--m-compact__action--MarginTop);
```

Comment:
An aside `--pf-c-data-list__action--MarginTop` was removed, but left here, although we need to add it back.

Reply:
updated